### DISCUSSION TITLE
Problems with PHP consuming large tasks

### DISCUSSION THREAD

**Discussion by runtu666** [rabbitmq/rabbitmq-server#11740] [Other] [answered]
### Is your feature request related to a problem? Please describe.

Using PHP as a RabbitMQ consumer, consumption tasks may be very time-consuming. The default heartbeat of RabbitMQ is 60s. Because of the way PHP works, it does not send heartbeats when consuming. When no heartbeat is received, RabbitMQ will think that the PHP consumer is offline and re-deliver the unacked task.

How to solve this problem

### Describe the solution you'd like

Can be processed normally, messages should not be delivered repeatedly

**Comment by michaelklishin** [comment, marked as answer]
@runtu666 please never again file issues for questions in repositories that have Discussions enabled.

Any traffic counts for a heartbeat with other clients. Use a client that does not have this limitation.

The heartbeat timeout can be configured, and there is an alternative that often is good enough. All of this has [been documented for many years](https://www.rabbitmq.com/docs/heartbeats).

**Reply by runtu666** [reply]
But this document does not have the heartbeat of php

**Reply by lukebakken** [reply]
We don't even know what PHP library you are using. If you are using `php-amqplib`, this issue has been discussed several times in that project:

https://github.com/search?q=repo%3Aphp-amqplib%2Fphp-amqplib+heartbeat+consume&type=issues
https://github.com/search?q=repo%3Aphp-amqplib%2Fphp-amqplib+heartbeat+consume&type=discussions
